### PR TITLE
[micromega] Switch from `Big_int` to ZArith.

### PR DESCRIPTION
**Pull request by ejgallego**
We replace micromega use of Num by Zarith.

### CODE REVIEW

**Comment by ejgallego** [comment]
This is basically in need of someone porting `microomega`, please feel free to do it.

**Comment by vbgl** [comment]
I can help regarding `micromega`. What’s the protocol for collaborating on a PR (esp. regarding force-pushing)?

**Comment by thery** [comment]
is it ok with everybody if we add a new dependency (gmp)?

**Comment by maximedenes** [comment]
> is it ok with everybody if we add a new dependency (gmp)?

Note that we are just replacing dependencies by others. I would much prefer depending on ZArith + GMP than on Num.

**Review by ppedrot — COMMENTED** [review]
No opinion.

**Comment by thery** [comment]
Num was not the home cooked suff by Valerie Menissier? gmp seems a bigger chunk

**Comment by maximedenes** [comment]
Here's what the `Num` README says:

This is a legacy library. It used to be part of the core OCaml distribution (in otherlibs/num) but is now distributed separately. New applications that need arbitrary-precision arithmetic should use the Zarith library (https://github.com/ocaml/Zarith) instead of the Num library, and older applications that already use Num are encouraged to switch to Zarith. Zarith delivers much better performance than Num and has a nicer API.

**Comment by ejgallego** [comment]
@thery has a point in that the transition won't be fully pain-free.

Also, I am not sure about the dependency of the frontend on `zarith`, but I guess we could factor that out. That being said, I do think the change is worth it.

**Comment by ejgallego** [comment]
Performance will be interesting, please add to the CI / Bench cool devs in that regard.

**Comment by thery** [comment]
If the motivation is micromega, I am not sure it will make much difference but I can be wrong.

**Comment by ejgallego** [comment]
@vbgl I need to push a couple of fixes, can I?

**Comment by ejgallego** [comment]
> @vbgl I need to push a couple of fixes, can I?

Ok pushed sorry, as I have to go and this has to build a full docker image.

**Comment by ejgallego** [comment]
Another docker version pushed.

File: dev/dune
Comment (SkySkimmer) [inline dev/dune]: Why's this included?

Comment (ejgallego) [reply dev/dune]: No special reason other than making my own dev setup work properly.

**Comment by ejgallego** [comment]
It seems 32 bit docker will need more work for the CI to run :(

**Review by ejgallego — COMMENTED** [review]
@vbgl I am surprised this is enough, in fact, the core of Coq depends on `ZArith` now so it should be added to the base set of packages. Anyways, I suggest you use the Dune build if you wanna start working on `micromega` soon.

**Review by Zimmi48 — APPROVED** [review]

**Comment by ejgallego** [comment]
@vbgl this is stable from my part, we should be able to run the CI once you push the micromega part. but note the hacks I had to do for the 32 bit built. This indeed raises a question on whether depending on Zarith is too inconvenient.

**Comment by vbgl** [comment]
I pushed something that does not work ;-) It type-checks but there are run-time failures. In case anyone wants to give a look.

**Comment by ejgallego** [comment]
Here you have the problem:
```
utop # Big_int_Z.(gcd_big_int zero_big_int unit_big_int);;
Exception: Division_by_zero.

utop # Big_int.(gcd_big_int zero_big_int unit_big_int);;
- : Big_int.big_int = <big_int 1>
```
wonder if that's worth reporting. Anyways, the fix should be easy, building now here.

**Comment by ejgallego** [comment]
Ok, fixed a bunch of things, pushed again. The test suite needs and update of the `csdp` cache I think.

File: plugins/micromega/sos.ml
Comment (vbgl) [inline plugins/micromega/sos.ml]: This fix is wrong. One is absorbing for GCD. I’ll report the bug upstream. Thanks for tracking it down.

Comment (vbgl) [reply plugins/micromega/sos.ml]: https://github.com/ocaml/Zarith/pull/39

Comment (ejgallego) [reply plugins/micromega/sos.ml]: Oh indeed, you want `gcd 0 n = n`. I suggest we add our own gcd version then for now.

I don't understand what is going it the make-based build system :S

Comment (ejgallego) [reply plugins/micromega/sos.ml]: In fact I suggest we do _à la_ mathcomp and provide a `\biggcd = gcd_list` that correctly handles this.

Comment (ejgallego) [reply plugins/micromega/sos.ml]: Rebasing to solve conflict and amending the bad fix.

**Comment by ejgallego** [comment]
@vbgl the problem with the "make" build is that you introduced a dependency on `Sos_lib` which was private before and you didn't update the proper `.mlpack` file.

Indeed, there is not easy choice with make due to packing issues and the hack currently used [it is a bit long to explain] so I will push a fixup to your commit removing the use of `Sos_lib` in the plugin.

**Comment by ejgallego** [comment]
Ok this should build now.

**Comment by ejgallego** [comment]
This is mainly pending on some test-suite problems:
- micromega/csdp interaction seems broken: I'm afraid this will need an expert.
- coq_makefile: problems linking `zarith`, needs an expert too.

**Comment by maximedenes** [comment]
@ejgallego does this still "needs: merge of dependency"? If yes, what is the dependent PR?

**Comment by ejgallego** [comment]
Indeed the dependency was merged.

**Comment by maximedenes** [comment]
@ejgallego what's the current status of this?

**Comment by vbgl** [comment]
The issue in `zarith` has been fixed upstream.

A few overlays are missing I guess.

**Comment by ejgallego** [comment]
> A few overlays are missing I guess.

Still a few problems with the test suite too:
```
FAILURES
    micromega/bertot.v...Error! (should be accepted)
    micromega/bertot.v...could not be checked (Error!)
    micromega/example.v...Error! (should be accepted)
    micromega/example.v...could not be checked (Error!)
    micromega/example_nia.v...Error! (should be accepted)
    micromega/example_nia.v...could not be checked (Error!)
    micromega/non_lin_ci.v...Error! (should be accepted)
    micromega/non_lin_ci.v...could not be checked (Error!)
    micromega/qexample.v...Error! (should be accepted)
    micromega/qexample.v...could not be checked (Error!)
    micromega/rexample.v...Error! (should be accepted)
    micromega/rexample.v...could not be checked (Error!)
    micromega/zomicron.v...Error! (should be accepted)
    micromega/zomicron.v...could not be checked (Error!)
```

**Comment by vbgl** [comment]
Indeed, this breaks the test-suite. Do you understand what happens, Emilio?

**Comment by ejgallego** [comment]
> Indeed, this breaks the test-suite. Do you understand what happens, Emilio?

No idea, I had a look at it but I don't remember now.

**Comment by maximedenes** [comment]
@ejgallego do you still plan to get this ready for 8.10? Otherwise maybe we should bump the milestone.

File: plugins/extraction/big.ml
Comment (liyishuai) [inline plugins/extraction/big.ml]: Have you tested this module? OCaml `Z` conflicts with Coq `Z` per https://github.com/coq/coq/pull/8252#discussion_r219342270. Given that PR we don't quite need to fix `Big` here.

**Comment by ejgallego** [comment]
It looks like this won't be ready soon from my part, pity it was fairly close to be completed tho.

**Comment by maximedenes** [comment]
> it was fairly close to be completed tho

Can you post here what remains to be done, so that others can contribute?

**Comment by ejgallego** [comment]
See https://github.com/coq/coq/pull/8743#issuecomment-471967613

File: interp/notation.ml
Comment (JasonGross) [inline interp/notation.ml]: The logic here is flipped.  The true case is for `i` odd

File: plugins/micromega/certificate.ml
Comment (JasonGross) [inline plugins/micromega/certificate.ml]: Merge conflict cruft

File: plugins/micromega/certificate.ml
Comment (JasonGross) [inline plugins/micromega/certificate.ml]: Merge conflict cruft

File: plugins/micromega/certificate.ml
Comment (JasonGross) [inline plugins/micromega/certificate.ml]: Merge conflict cruft

**Review by JasonGross — COMMENTED** [review]
I have not looked through everything, but perhaps the test-suite fails is caused by the flipped logic I found?

File: interp/notation.ml
Comment (ejgallego) [reply interp/notation.ml]: Fixed, thanks @JasonGross ; note what with the new Z api many things can be implemented better.

**Comment by ejgallego** [comment]
Ok, thanks for the review @JasonGross ; I've resurrected this just in case to see what's going on.

**Comment by ejgallego** [comment]
> Ok, thanks for the review @JasonGross ; I've resurrected this just in case to see what's going on.

Problems with the test suite in micromega still persist; they are some tricky spots so indeed someone has to take a careful look to see where the problem could be, maybe @fajb has some idea?

File: plugins/micromega/certificate.ml
Comment (ejgallego) [inline plugins/micromega/certificate.ml]: This code was strange before, we should check.

**Comment by ejgallego** [comment]
Seems close, but the test suite for micromega is still creating problems; I think the changes in exponentiation are likely to be the root cause, I'm unsure of what the original code meant when using `num_power` (cc: @fajb )

**Comment by JasonGross** [comment]
@ejgallego do you mean `power_num`?  The translation looks correct to me; the version in `num` errors if the exponent is not an integer, so assuming that there is no handler of that error, doing `floorZ` instead should be fine.  (I could only find one interesting use of `power_num`, though, so perhaps I missed something?)

**Comment by ejgallego** [comment]
I've splitted the parts that are ready to merge to #11742 , once that PR is merged this shall become only about the micromega part.

**Comment by fajb** [comment]
I have no idea of the origin of the problem.
@ejgallego , that was a bold move to port the code straight ahead. Now, this is very hard to debug. Maybe a smoother path would have been to keep a compatibility layer and implement it with both Num and ZArith. 
That would have make possible to execute the code with both of them and pinpoint the decrepancy.
 I guess this is too late now.

**Comment by ejgallego** [comment]
@fajb I've added a compatibility layer, I need to push it before the migration, but that should be doable soon.

**Comment by ejgallego** [comment]
Compat layer more or less done, still we have a problem somewhere but it should be easy now to pin down.

**Review by maximedenes — COMMENTED** [review]
@ejgallego what's the status of this PR? Should it be closed? Turned into a draft? It doesn't seem ready for review.

**Comment by ejgallego** [comment]
> @ejgallego what's the status of this PR? Should it be closed? Turned into a draft? It doesn't seem ready for review.

@maximedenes this PR deals only with the micromega part, it depends on #11742 for the infrastructure. Turned into a draft for now.

**Comment by ejgallego** [comment]
@fajb this PR is now minimal and should be easy to debug [start with the first commit, however it doesn't report anything strange]

It could well be some serialization problem?

**Comment by fajb** [comment]
I am looking into it.

> It could well be some serialization problem?
It does not seem so. Disabling caches does not solve the problem...

File: plugins/micromega/numCompat.ml
Comment (ejgallego) [inline plugins/micromega/numCompat.ml:307]: :face_with_head_bandage: :facepalm: :man_facepalming: :woman_facepalming: :woman_astronaut:

**Comment by ejgallego** [comment]
Ok one worrying failure:
```
assertion failure at gcd
File "./src/Helpers/ModArith.v", line 57, characters 2-6:
```

File: plugins/micromega/numCompat.ml
Comment (ejgallego) [reply plugins/micromega/numCompat.ml:309]: Updated, and added some more debug code.

**Comment by ejgallego** [comment]
Gonna run a CI with #13002 added and without the debug code so we can bench and also get a complete CI.

**Comment by ejgallego** [comment]
For perennial, the problem is:
```
assertion failure at gcd, [18446744073709551616] vs [-18446744073709551616]
File "./src/Helpers/ModArith.v", line 57, characters 2-6:
```
so likely some silly typo again.

**Comment by ejgallego** [comment]
Ok, this seems finally ready for review and bench. Note that the issue in https://github.com/ocaml/Zarith/issues/58 created some problems in testing, but none in practice; anyways we've added a workaround.

Debug code lives in https://github.com/ejgallego/coq/tree/zarith%2Bdebug

**Comment by fajb** [comment]
There is still a problem with the .csdp.cache of the test-suite?
I don't remember how this is handled.
Is it supposed to be already there or regenerated?

**Comment by vbgl** [comment]
> I don't remember how this is handled.
> Is it supposed to be already there or regenerated?

It should be already there, named: `test-suite/.csdp.cache.test-suite`

**Comment by ejgallego** [comment]
The test suite is actually failing due to out of memory! It worked in my machine.

Maybe some `zarith` bug ? Seems pretty weird indeed.

**Comment by ejgallego** [comment]
> It should be already there, named: `test-suite/.csdp.cache.test-suite`

@fajb I updated that file, things were working fine in my machine, but indeed the memory consumption is strange as our test suite has noticed.

Bench is over by the way:
```
┌─────────────────────────────┬─────────────────────────┬─────────────────────────────────────────────┬─────────────────────────────────────────────┬───────────────────────────────┬───────────────────┐
│                             │      user time [s]      │                 CPU cycles                  │              CPU instructions               │     max resident mem [KB]     │    mem faults     │
│                             │                         │                                             │                                             │                               │                   │
│                package_name │     NEW     OLD PDIFF   │               NEW               OLD PDIFF   │               NEW               OLD PDIFF   │        NEW        OLD PDIFF   │  NEW  OLD PDIFF   │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                 coq-bignums │   57.52   58.14 -1.07 % │      159473439997      160743140667 -0.79 % │      215256412407      215405481574 -0.07 % │     472000     471360 +0.14 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                    coq-hott │  313.44  315.87 -0.77 % │      852125813994      854152466163 -0.24 % │     1372337168651     1372139577735 +0.01 % │     635796     676888 -6.07 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                 coq-coqutil │   59.07   59.49 -0.71 % │      164066630624      165249278506 -0.72 % │      204867373818      207850670552 -1.44 % │     528876     533292 -0.83 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│             coq-fiat-crypto │ 4933.56 4964.10 -0.62 % │    13703805770489    13780276215807 -0.55 % │    23632356922890    23808769949152 -0.74 % │    2326364    2326560 -0.01 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                coq-coqprime │   80.62   81.02 -0.49 % │      221772219356      222290123321 -0.23 % │      310222771856      310606456790 -0.12 % │     768808     768732 +0.01 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│              coq-coquelicot │   69.59   69.91 -0.46 % │      191199255689      190908716758 +0.15 % │      245176134888      245136030621 +0.02 % │     585232     587252 -0.34 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│       coq-mathcomp-fingroup │   47.92   48.00 -0.17 % │      133034489372      132729052337 +0.23 % │      183737227625      183653953473 +0.05 % │     486296     486292 +0.00 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                coq-bedrock2 │  220.50  220.74 -0.11 % │      609995514725      610764130742 -0.13 % │     1009815212129     1010897419381 -0.11 % │    1087348    1087252 +0.01 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                         coq │  542.33  542.88 -0.10 % │     1515870429882     1517190268687 -0.09 % │     2073129885186     2075059428096 -0.09 % │     592384     592152 +0.04 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│              coq-verdi-raft │ 1253.31 1254.45 -0.09 % │     3491485199898     3492305794495 -0.02 % │     4896883656609     4898588982213 -0.03 % │     934516     934728 -0.02 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│          coq-mathcomp-field │  290.97  291.20 -0.08 % │      810371423923      809839193393 +0.07 % │     1249424876124     1249297149216 +0.01 % │     825928     825676 +0.03 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│  coq-performance-tests-lite │  543.97  544.36 -0.07 % │     1520691914914     1520116420525 +0.04 % │     2795268552108     2796442877783 -0.04 % │     747332     747384 -0.01 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│      coq-mathcomp-odd-order │ 1070.16 1070.87 -0.07 % │     2980695269124     2981616599146 -0.03 % │     4750209131583     4750233816415 -0.00 % │     992036     992080 -0.00 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│            coq-fiat-parsers │  643.44  643.84 -0.06 % │     1802603676169     1802453523131 +0.01 % │     2717331471990     2717370622014 -0.00 % │    2998104    2997984 +0.00 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│               coq-fiat-core │  105.45  105.51 -0.06 % │      301576060554      300758643181 +0.27 % │      401371168791      401333791593 +0.01 % │     487060     487068 -0.00 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│       coq-mathcomp-solvable │  172.38  172.44 -0.03 % │      479703885177      479233959314 +0.10 % │      677556062495      677468542362 +0.01 % │     691832     691720 +0.02 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│       coq-engine-bench-lite │  380.02  380.05 -0.01 % │     1069023511982     1068285842923 +0.07 % │     1838925945920     1838247888520 +0.04 % │    3508668    3508732 -0.00 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│               coq-fourcolor │ 2518.64 2518.32 +0.01 % │     7022901241797     7019509600213 +0.05 % │    11447737865163    11447223653710 +0.00 % │     774876     775024 -0.02 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│             coq-lambda-rust │ 1262.98 1262.76 +0.02 % │     3515011128111     3513461858344 +0.04 % │     4949338217145     4946675717171 +0.05 % │    1159380    1159368 +0.00 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│ coq-rewriter-perf-SuperFast │ 1057.38 1056.95 +0.04 % │     2940976697968     2940895197848 +0.00 % │     4629379715349     4629740954555 -0.01 % │    1070804    1070672 +0.01 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                  coq-geocoq │ 1469.75 1469.07 +0.05 % │     4102093064533     4097309838052 +0.12 % │     6222263366825     6222091958462 +0.00 % │    1190120    1146084 +3.84 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                coq-rewriter │  610.51  610.22 +0.05 % │     1694597080132     1692249153052 +0.14 % │     2583731738860     2583918291568 -0.01 % │    1036784     987712 +4.97 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                    coq-corn │ 1707.32 1705.26 +0.12 % │     4754184603733     4750701788800 +0.07 % │     7278243980988     7278474379110 -0.00 % │     855684     855604 +0.01 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                   coq-flocq │  159.48  159.28 +0.13 % │      443219035633      442805735154 +0.09 % │      552990623634      553358794765 -0.07 % │     822204     822128 +0.01 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                 coq-unimath │ 3993.34 3987.00 +0.16 % │    11111629365061    11095643671726 +0.14 % │    21160840068546    21166523184354 -0.03 % │    1113380    1116692 -0.30 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                coq-compcert │  663.84  662.62 +0.18 % │     1848990291336     1848235883695 +0.04 % │     2555820339465     2555802379899 +0.00 % │    1131576    1131244 +0.03 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│        coq-mathcomp-algebra │  140.40  140.14 +0.19 % │      389907236902      389371135024 +0.14 % │      503642194141      503577259557 +0.01 % │     562908     564560 -0.29 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                   coq-verdi │  120.83  120.51 +0.27 % │      332438231484      332729584726 -0.09 % │      413031222258      413028766973 +0.00 % │     584464     585620 -0.20 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│      coq-mathcomp-character │  160.69  160.25 +0.27 % │      446290493749      445446063798 +0.19 % │      618941348991      618925419293 +0.00 % │     734624     735040 -0.06 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│                   coq-color │  618.37  616.30 +0.34 % │     1723501371646     1718757700920 +0.28 % │     2190345082800     2190190676093 +0.01 % │    1455192    1455288 -0.01 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│      coq-mathcomp-ssreflect │   50.39   50.14 +0.50 % │      138115038496      137633047582 +0.35 % │      176210590083      176150819239 +0.03 % │     521688     521560 +0.02 % │    0    0  +nan % │
├─────────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼───────────────────┤
│            coq-math-classes │  190.48  189.52 +0.51 % │      530678395166      529859918698 +0.15 % │      700214855833      700186292794 +0.00 % │     504956     504808 +0.03 % │    0    0  +nan % │
└─────────────────────────────┴─────────────────────────┴─────────────────────────────────────────────┴─────────────────────────────────────────────┴───────────────────────────────┴───────────────────┘
```
No significative change IMO.

**Comment by fajb** [comment]
@ejgallego the bench looks good. 
commit https://github.com/coq/coq/pull/8743/commits/229350bda6dba57e896ccb2d84c66c8fc4f33a86 has a fresh `.csdp.cache`. It  works even when csdp is uninstalled. 
I have also added `zarith` to `merlin.in` and `make clean` removes .csdp.cache from the test-suite.

**Comment by silene** [comment]
> the bench looks good

Should we worry that the memory footprint of `coq-geocoq` and `coq-rewriter` has increased by 5%?

**Comment by ejgallego** [comment]
> Should we worry that the memory footprint of `coq-geocoq` and `coq-rewriter` has increased by 5%?

If not reproducible likely not; I don't know how to trigger a bench for 2 development only now tho.

**Comment by fajb** [comment]
@ejgallego No idea. I have never looked at memory consumption.
My feeling was that zarith was designed to be GC friendly.

**Comment by ejgallego** [comment]
> @ejgallego the bench looks good.
> commit [229350b](https://github.com/coq/coq/commit/229350bda6dba57e896ccb2d84c66c8fc4f33a86) has a fresh `.csdp.cache`. It works even when csdp is uninstalled.
> I have also added `zarith` to `merlin.in` and `make clean` removes .csdp.cache from the test-suite.

Thanks, I've launched the CI.

> @ejgallego No idea. I have never looked at memory consumption.
> My feeling was that zarith was designed to be GC friendly.

Indeed it seems the OOM could maybe a problem with the csdp cache? Let see what the re-run says.

File: plugins/micromega/numCompat.ml
Comment (soraros) [inline plugins/micromega/numCompat.ml:126]: There is `Q.minus_one`

Comment (ejgallego) [reply plugins/micromega/numCompat.ml]: Good catch, thanks!

**Comment by ejgallego** [comment]
@fajb I think we are close to a fixpoint here, would you like to assign?

**Review by fajb — APPROVED** [review]

**Comment by fajb** [comment]
@ejgallego fixpoint, eventually! I'll merge as soon as the CI passes.

**Comment by ejgallego** [comment]
> @ejgallego fixpoint, eventually! I'll merge as soon as the CI passes.

Failure on coq_tools is spurious [failing everywhere] and not related to this PR.

**Comment by fajb** [comment]
@ejgallego well done.

**Comment by ejgallego** [comment]
Thanks @fajb !

**Pull request merged**
